File: src/leetCodeProblems/TwoPointers/StringReverse344.java
package leetCodeProblems.TwoPointers;

/**
 * LeetCode Problem Link - https://leetcode.com/problems/reverse-string/
 * */

public class StringReverse344 {

	public void reverseString(char[] s) {
        int lastIndex = s.length - 1;

        for(int i = 0;i < s.length/2;i++) {
            char temp = s[lastIndex];
            s[lastIndex] = s[i];
            s[i] = temp;
            lastIndex--;
        }
    }
}

File: src/leetCodeProblems/TwoPointers/StringReverse344.go
package main

/*
- Leetcode - https://leetcode.com/problems/reverse-string/description/
- Time - O(n)
- Space - O(1)
*/
func reverseString(s []byte) {
	first, last := 0, len(s)-1

	for first < last {
		temp := s[first]
		s[first] = s[last]
		s[last] = temp
		first++
		last--
	}
}

// func main() {
// 	input := []byte{'h', 'e', 'h', 'e', 'l', 'l', 'o'}
// 	reverseString(input)
// 	log.Println(string(input))
// }
